frontend/pkg/frontend/middleware_validatesubscription_test.go

```diff
@@ -155,9 +155,21 @@ func TestMiddlewareValidateSubscription(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			dbClient := database.NewCache()
+			middleware := NewSubscriptionStateMuxValidator(dbClient)
 
 			if tt.cachedState != "" {
-				cache.SetSubscription(subscriptionId, &arm.Subscription{State: tt.cachedState})
+				if err := dbClient.SetSubscriptionDoc(context.Background(), &database.SubscriptionDocument{
```

Comment:
Maybe we could move &database.SubscriptionDocument to a variable to improve readability.